### ISSUE TITLE
Flaky test: KituraWebSocketTests.ProtocolErrorTests testBinaryAndTextFrames

### ISSUE THREAD

**Issue by glbrntt** [IBM-Swift/Kitura-WebSocket-NIO#63]
`testBinaryAndTextFrames` is failing because of an over fulfilled expectation.

Commit: 78d25e2
Swift: 5.1.2

```
$ swift test --sanitize=address -c release -Xswiftc -enable-testing
```

Relevant output:
```
Test Case '-[KituraWebSocketTests.ProtocolErrorTests testBinaryAndTextFrames]' started.                                                                                                                                                                                                                                                                                     
2019-12-05 14:08:03.181 xctest[29130:2571787] *** Assertion failure in -[XCTestExpectation fulfill], /Library/Caches/com.apple.xbs/Sources/XCTest/XCTest-15509/Sources/XCTestFramework/Async/XCTestExpectation.m:79                                                                                                                                                         
2019-12-05 14:08:03.185 xctest[29130:2571787] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'API violation - multiple calls made to -[XCTestExpectation fulfill] for ConnectionTests:22[0].'
*** First throw call stack:                                                                                                                                                           
(                                                                                                                                                                                     
        0   CoreFoundation                      0x00007fff34ba68ab __exceptionPreprocess + 250                                                                                                                                                                                                                                                                              
        1   libobjc.A.dylib                     0x00007fff6af50805 objc_exception_throw + 48                                                                                                                                                                                                                                                                                
        2   CoreFoundation                      0x00007fff34bcfd10 +[NSException raise:format:arguments:] + 88                                                                                                                                                                                                                                                              
        3   Foundation                          0x00007fff372c8221 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 191                                                                                                                                                                                                                    
        4   XCTest                              0x0000000105b8ae05 -[XCTestExpectation fulfill] + 652                                                              
        5   KituraWebSocketPackageTests         0x000000010ca609f0 $s20KituraWebSocketTests010ConnectionD0C22testDisconnectCallbackyyFySo17XCTestExpectationCcfU_y3NIO7Channel_p_AG10ByteBufferVtcfU_ + 960
        6   KituraWebSocketPackageTests         0x000000010ca61bd6 $s20KituraWebSocketTests010ConnectionD0C22testDisconnectCallbackyyFySo17XCTestExpectationCcfU_y3NIO7Channel_p_AG10ByteBufferVtcfU_TA + 38
        7   KituraWebSocketPackageTests         0x000000010ca80c2b $s20KituraWebSocketTests0bC14MessageHandlerC11channelRead7context4datay3NIO07ChannelF7ContextC_AG6NIOAnyVtF + 7707
        8   KituraWebSocketPackageTests         0x000000010cb3adda $s3NIO21ChannelHandlerContextC06invokeB4Read33_EEC863903996E9F191EBAFEB0FB0DFDDLLyyAA6NIOAnyVF + 154
        9   KituraWebSocketPackageTests         0x000000010cb2b6c0 $s3NIO21ChannelHandlerContextC04fireB4ReadyyAA6NIOAnyVF + 64
        10  KituraWebSocketPackageTests         0x000000010cdc755e $s12NIOWebSocket03WebB12FrameDecoderC6decode7context6buffer3NIO13DecodingStateOAG21ChannelHandlerContextC_AG10ByteBufferVztKF + 798
        11  KituraWebSocketPackageTests         0x000000010cdc7799 $s12NIOWebSocket03WebB12FrameDecoderC3NIO013ByteToMessageE0AadEP6decode7context6bufferAD13DecodingStateOAD21ChannelHandlerContextC_AD0G6BufferVztKFTW + 41
        12  KituraWebSocketPackageTests         0x000000010cb77b07 $s3NIO20ByteToMessageHandlerC10decodeLoop33_F2A740607FEBA425AF6F8C49A24C0AD7LL7context0F4ModeAA10B2MDBufferAELLV22BufferProcessingResultOAA07ChannelE7ContextC_AC06DecodeR0AELLOyx_GtKFAA13DecodingStateOxz_AA0bU0VztKXEfU_ + 151
        13  KituraWebSocketPackageTests         0x000000010cb76cf6 $s3NIO20ByteToMessageHandlerC10decodeLoop33_F2A740607FEBA425AF6F8C49A24C0AD7LL7context0F4ModeAA10B2MDBufferAELLV22BufferProcessingResultOAA07ChannelE7ContextC_AC06DecodeR0AELLOyx_GtKF + 3606
        14  KituraWebSocketPackageTests         0x000000010cb7919f $s3NIO20ByteToMessageHandlerC11channelRead7context4datayAA07ChannelE7ContextC_AA6NIOAnyVtF + 783
        15  KituraWebSocketPackageTests         0x000000010cb3adda $s3NIO21ChannelHandlerContextC06invokeB4Read33_EEC863903996E9F191EBAFEB0FB0DFDDLLyyAA6NIOAnyVF + 154
        16  KituraWebSocketPackageTests         0x000000010cb3ae1f $s3NIO21ChannelHandlerContextC06invokeB4Read33_EEC863903996E9F191EBAFEB0FB0DFDDLLyyAA6NIOAnyVF + 223
        17  KituraWebSocketPackageTests         0x000000010cae5e61 $s3NIO23BaseStreamSocketChannelC08readFromD0AA0bdE0C10ReadResultOyx_GyKFAA0D0C_Tg5 + 4737
        18  KituraWebSocketPackageTests         0x000000010cad4647 $s3NIO17BaseSocketChannelC9readable0019_9C3D3FC8F9F3D83A05G12F6063F6AE958LLAC15ReadStreamStateAELLOyx_GyFAA0C0C_Tg5 + 1431
        19  KituraWebSocketPackageTests         0x000000010cbb6bed $s3NIO19SelectableEventLoopC06handleC0_7channelyAA08SelectorC3SetV_xtAA0B7ChannelRzlFAA06SocketI0C_Tg5Tf4nnd_n + 4525
        20  KituraWebSocketPackageTests         0x000000010cc688ab $s3NIO8SelectorC9whenReady8strategy_yAA0B8StrategyO_yAA0B5EventVyxGKXEtKFAA15NIORegistrationO_Tg5018$s3NIO19Selectableg26LoopC3runyyKFyyKXEfU_yAA08b8C0VyAA15H7OGXEfU_AA0kgL0CTf1ncn_n + 2635
        21  KituraWebSocketPackageTests         0x000000010cba42dd $s3NIO19SelectableEventLoopC3runyyKFyyKXEfU_ + 141                                                                                                                                                                                                                                                       
        22  KituraWebSocketPackageTests         0x000000010cbb2691 $s3NIO19SelectableEventLoopC3runyyKFyyKXEfU_TA + 17                                                                                                                                                                                                                                                      
        23  KituraWebSocketPackageTests         0x000000010cb9842c $ss5Error_pIgzo_ytsAA_pIegrzo_TR + 12
        24  KituraWebSocketPackageTests         0x000000010cbb26b1 $ss5Error_pIgzo_ytsAA_pIegrzo_TRTA + 17
        25  KituraWebSocketPackageTests         0x000000010cba0a9c $s3NIO19withAutoReleasePool33_D5D78C61B22284700B9BD1ACFBC25157LLyxxyKXEKlFxyKXEfU_yt_Tg5 + 12
        26  KituraWebSocketPackageTests         0x000000010cbb26d1 $s3NIO19withAutoReleasePool33_D5D78C61B22284700B9BD1ACFBC25157LLyxxyKXEKlFxyKXEfU_yt_Tg5TA + 17
        27  libswiftObjectiveC.dylib            0x00007fff6bf67f0e $s10ObjectiveC15autoreleasepool8invokingxxyKXE_tKlF + 46
        28  KituraWebSocketPackageTests         0x000000010cba3567 $s3NIO19SelectableEventLoopC3runyyKF + 759                                           
        29  KituraWebSocketPackageTests         0x000000010cba6c5f $s3NIO27MultiThreadedEventLoopGroupC014setupThreadAnddE033_D5D78C61B22284700B9BD1ACFBC25157LL4name11initializerAA010SelectabledE0CSS_yAA9NIOThreadCctFZyAKcfU_ + 575
        30  KituraWebSocketPackageTests         0x000000010cbc2149 $s3NIO27MultiThreadedEventLoopGroupC014setupThreadAnddE033_D5D78C61B22284700B9BD1ACFBC25157LL4name11initializerAA010SelectabledE0CSS_yAA9NIOThreadCctFZyAKcfU_TA + 41
        31  KituraWebSocketPackageTests         0x000000010cc303a8 $s3NIO9NIOThreadCIegg_ACytIegnr_TR + 40                                                                                                                                                                                                                                                                  
        32  KituraWebSocketPackageTests         0x000000010cbc2161 $s3NIO9NIOThreadCIegg_ACytIegnr_TRTA + 17                                                                         
        33  KituraWebSocketPackageTests         0x000000010ccb1eae $s3NIO9NIOThreadC11spawnAndRun4name12detachThread4bodyySSSg_SbyACctFZSvSgSvcfU_ + 558               
        34  libsystem_pthread.dylib             0x00007fff6c4c2e65 _pthread_start + 148                                                                                               
        35  libsystem_pthread.dylib             0x00007fff6c4be83b thread_start + 15                                                                                                                                                                                                                                                                                        
)                                                                                                                                                                                                                                                                                                                                                                           
libc++abi.dylib: terminating with uncaught exception of type NSException                                                                                                                                                                                                                                                                                                    
Exited with signal code 6 
```

**Comment by harish1992** [comment]
fixed in PR #62

**Comment by harish1992** [comment]
Hi @glbrntt, PR #62 fixes this issue, can this be closed now? Thanks.
